### DISCUSSION TITLE
Rename movies and series using custom templates and TMDb data for easy indexing and clear filenames

### DISCUSSION THREAD

**Discussion by XTorLukas** [sunecz/Media-Downloader#8] [Ideas]
Proposal for renaming movie and series titles using a custom template and data retrieval from www.themoviedb.org, including title, year, tmdb-ID, etc., for example:

- [tmdbTvShowName] SXXEXX [tmdbEpisodeName] {tmdb-[tmdbID]}.ext
- [tmdbMovieName] ([tmdbYear]) {tmdb-[tmdbID]}.ext

This approach facilitates subsequent indexing and organization.

**Comment by sunecz** [comment]
Hello, thank you for the propsal! I'll look into it in more detail. It depends on how good is the website in searching and normalizing data, but it seems feasible.

**Reply by XTorLukas** [reply]
Hello, the TMDb website uses an [API](https://developer.themoviedb.org/reference/intro/getting-started) interface for access, where each user has their own API key.

So the actual tracking of information should not be difficult to implement. I personally use my own pseudocode in C# and python for renaming. I don't know Java, only C/C++ and C#.

I think others, would appreciate building it directly into MD.
I like your work, keep it up.

**Reply by sunecz** [reply]
Hello, thank you for the information! I'll look into it, but please expect that this functionality will be present in the 00.02.10 development versions at the earliest.
